src/panel/category/CategoryPanel.jsx

```diff
@@ -45,11 +44,25 @@ export default class CategoryPanel extends React.Component {
     window.execOnMapLoaded(() => { this.fitMapAndFetch(); });
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps) {
+    const { bbox, poiFilters } = this.props;
+
     const panelContent = document.querySelector('.panel-content');
     if (panelContent) {
       panelContent.scrollTop = 0;
     }
+
+    // Check for a new, or changed poiFilter
+    for (const key in poiFilters) {
```

Comment:
This assumes the same keys are present in both objects. It's the case for now just because there is a single render point for `CategoryPanel` which always pass `category` and `query`. But I'm quite sure in the future we'll have need for something more generic like a shallowCompare function. Let's not forget it then.